django/db/models/lookups.py

```diff
@@ -94,7 +94,13 @@ def process_rhs(self, compiler, connection):
             value = self.apply_bilateral_transforms(value)
             value = value.resolve_expression(compiler.query)
         if hasattr(value, 'as_sql'):
-            return compiler.compile(value)
+            sql, params = compiler.compile(value)
+            # Ensure expression is wrapped in parentheses to respect operator
+            # precedence but avoid double wrapping as it can be misinterpreted
+            # on some backends (e.g. subqueries on SQLite).
+            if sql and sql[0] != '(':
+                sql = '(%s)' % sql
```

Comment:
Is there any way to prevent this? I'm actually running into a PostgreSQL syntax error because of it.

I have these two custom functions:

```python
from django.db.models.expressions import Func


class Any(Func):
    function = "ANY"


class Array(Func):
    function = "ARRAY"
```

And I'm trying to run a query similar to this:

```python
Foo.objects.filter(
    pk=Any(
        Array(
            Foo.objects.filter(...).values("pk")[:1000]
        )
    )
).delete()
```

Before this change, it produced SQL like this:

```sql
WHERE V0."id" = ANY(ARRAY(...));
```

After this change I get this which isn't valid:

```sql
WHERE V0."id" = (ANY(ARRAY(...)));
```

```
ERROR:  syntax error at or near "ANY"
LINE 1: ... V0."id" FROM "foo" V0 WHERE V0."id" = (ANY(ARRAY(...
```